### ISSUE TITLE
Vercel deployment: Unexpected token < in JSON at position 0

### ISSUE THREAD

**Issue by AnnieTaylorCHEN** [sveltejs/kit#3950]
### Describe the bug

Previous build was totally alright. It's a SvelteKit + DatoCMS app. 

So from this main branch I opened a new branch, changed three lines of text (note not programmatically, just plain text), then I deployed it to vercel like usual but getting 
`Unexpected token < in JSON at position 0`

So far local dev, build, preview all worked fine. Vercel build logs shows it's successful and no errors. 

I have since tried:
- ran npm update to update to SvelteKit v1.0.0-next.271
- put headers: { Accept: 'application/json' } as second arg in the fetch() in load
- tried to switch to adapter-auto
- redeploy without build cache
but none of the above worked. 

### Reproduction

Unfortunately it's private so I can't share it. 

### System Info

```shell
System:
    OS: macOS 12.1
    CPU: (8) arm64 Apple M1
    Memory: 142.58 MB / 16.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 14.18.2 - ~/.nvm/versions/node/v14.18.2/bin/node
    Yarn: 1.22.17 - ~/.nvm/versions/node/v14.18.2/bin/yarn
    npm: 8.5.0 - ~/.nvm/versions/node/v14.18.2/bin/npm
  Browsers:
    Chrome: 98.0.4758.80
    Firefox: 97.0
    Safari: 15.2
  npmPackages:
    @sveltejs/adapter-auto: next => 1.0.0-next.17 
    @sveltejs/adapter-vercel: next => 1.0.0-next.43 
    @sveltejs/kit: next => 1.0.0-next.271 
    svelte: ^3.46.0 => 3.46.4
    vite v2.8.3 (from Vercel log)
```

### Severity

blocking all usage of SvelteKit

**Comment by Conduitry** [comment]
If you can't share the project this is happening in, you will need to create another reproduction of the issue that you can share.

**Comment by AnnieTaylorCHEN** [comment]
Hi, I found the issue finally! It was due to that I forgot to enable all the env variables on Vercel for "preview". Since Main branch has "production" env so it always builds without problem. The new branch has preview link and it lacks the env there, which resulted in the error. 😊 

Another trick to debug this is to use check functions-> Realtime logs. By refreshing your preview link you shall see some errors pop up here.
